src/pages/workspaces/workspace/notebooks/NotebookLauncher.js

```diff
@@ -0,0 +1,150 @@
+import _ from 'lodash/fp'
+import { Fragment } from 'react'
+import { div, h } from 'react-hyperscript-helpers'
+import { icon, spinner } from 'src/components/icons'
+import { TopBar } from 'src/components/TopBar'
+import { Rawls, Leo } from 'src/libs/ajax'
+import { getBasicProfile } from 'src/libs/auth'
+import { reportError } from 'src/libs/error'
+import * as Style from 'src/libs/style'
+import * as Nav from 'src/libs/nav'
+import * as Utils from 'src/libs/utils'
+import { Component } from 'src/libs/wrapped-components'
+
+
+const wait = ms => new Promise(resolve => setTimeout(resolve, ms))
+
+
+const styles = {
+  pageContainer: {
+    padding: '2rem'
+  },
+  step: {
+    container: {
+      display: 'flex',
+      alignItems: 'center',
+      lineHeight: '2rem',
+      margin: '0.5rem 0'
+    },
+    col1: {
+      flex: '0 0 30px'
+    },
+    col2: {
+      flex: 1
+    }
+  }
+}
+
+
+class NotebookLauncher extends Component {
+  constructor(props) {
+    super(props)
+
+    this.creator = getBasicProfile().getEmail()
+  }
+
+  async componentDidMount() {
+    const bucketName = await this.resolveBucketName()
+    this.setState({ bucketName })
+    const cluster = await this.startCluster()
+    await this.localizeNotebook(cluster)
+
+    const { name: workspaceName, notebookName } = this.props
+    window.location.href = `${cluster.clusterUrl}/notebooks/${workspaceName}/${notebookName}`
+  }
+
+  async resolveBucketName() {
+    const { namespace, name: workspaceName } = this.props
+
+    try {
+      const { workspace: { bucketName } } = await Rawls.workspace(namespace, workspaceName).details()
+      return bucketName
+    } catch (error) {
+      reportError('Error during Google bucket lookup', error)
```

Comment:
Be careful with the error flow. Catching the error here will cause this function's promise to resolve rather than reject, so the top-level logic will continue, even though it probably shouldn't. We should probably either 1. re-throw the exception here, or 2. not catch it at this level, and instead handle things uniformly at the top level.